control-agent/text-to-speech/speech-detection.py

```diff
@@ -162,11 +219,72 @@ def QnA(question):
             now = datetime.now()
             speakabletime = now.strftime("%H:%M")
             speech.speak("The time is, " + speakabletime)
+
         elif intent.intent_id == "Stop":
             break
+        elif intent.intent_id == "Selfie":
```

Comment:
In future, I'd pass each intent to a function. You could have a dictionary of function references

Reply:
Yes that's a good idea, I'll put in a card for a refactor